### DISCUSSION TITLE
Can we have utility from `tada` can return `enum` object

### DISCUSSION THREAD

**Discussion by tianyingchun** [0no-co/gql.tada#263] [Q&A] [answered]
i know for now tada support infer typings for `scalar` like 
```ts
export type Permissions = ReturnType<typeof graphql.scalar<'Permission'>>;

```

but if here have method to like `graphql-codegen` generated a enum object like
```ts
const Permissions =  graphql.scalar<'Permission'> or somethings 
// asume that the typings `Permission` have over 100+, we should make data automatically generated it that's better

```
we want get a vaiables `const x = Object.keys(Permissions)`

is there any suggestins?

**Comment by kitten** [comment, marked as answer]
Don't do this is my suggestion.

This may seem like a great idea at first, however, there are well documented resources on the drawbacks of enums in TypeScript. One could of course reach for const enums, but that explicitly excludes the case you're speaking about.
That's kind of for good reason. Unions of string literals are pretty forward-compatible, but enums are not and give off the impression that you can have exhaustive checks. But in the spirit of GraphQL backwards-compatibility enums are just not a great choice for this.

You **can** since a recent version replace the types that `gql.tada` uses for enums using the `scalars` option, e.g. you can override the type for a given enum type with a custom TypeScript type, so you can slot in TS enums you generate yourself.
However, from `gql.tada`'s perspective, we will never reach for code gen that generates runtime code, i.e. generating code that doesn't just compile away as types, especially since this kind of defeats the type-first nature of `gql.tada`.

Lastly, I repeatedly see patterns where people reach for GraphQL unions to make human-readable string mapping, instead of simply returning this from their GraphQL APIs as lists of object types, making their UI less GraphQL-driven and more client-driven.

**Comment by tianyingchun** [comment]
thanks for you explanation, haha agree
